app/Application/WalletCryptocurrencies/WalletCryptocurrenciesService.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Application\WalletCryptocurrencies;
+
+use App\Application\CryptoDataSource\CryptoDataSource;
+use App\Domain\Wallet;
+use Exception;
+
+class WalletCryptocurrenciesService
+{
+    /**
+     * @var CryptoDataSource
+     */
+    private CryptoDataSource $cryptoDataSource;
```

Comment:
The interface CryptoDataStorage should be use instead of CryptoDataSource as the information does not come from the external api but the persistent data from the api